edb/schema/types.py

```diff
@@ -1100,7 +1100,12 @@ def get_schema_class_displayname(cls) -> str:
     def as_type_delete_if_unused(
         self: CollectionTypeT,
         schema: s_schema.Schema,
-    ) -> sd.DeleteObject[CollectionTypeT]:
+    ) -> Optional[sd.DeleteObject[CollectionTypeT]]:
+        if not schema.get_by_id(self.id, default=None):
+            # this type was already deleted by some other op
+            # (probably alias types cleanup)
+            return None
+
```

Comment:
It is needed, `if_exists` does not cover it.

That's because there are cases where we don't toposort the deletes.
It happens when we are preparing a schema for alias compilation:
https://github.com/edgedb/edgedb/blob/b76ab0f8592fef525998519848e795d012d34108/edb/schema/expraliases.py#L216-L222

We could toposort the deletes, but this is faster and less error prone.